core/primitives/src/transaction.rs

```diff
@@ -348,6 +348,20 @@ pub struct ExecutionOutcome {
     /// NOTE: Should be the latest field since it contains unparsable by light client
     /// ExecutionStatus::Failure
     pub status: ExecutionStatus,
+    /// Execution metadata, versioned
+    pub metadata: ExecutionMetadata,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, PartialEq, Clone, Eq, Debug)]
+pub enum ExecutionMetadata {
+    // V1: Empty Metadata
+    ExecutionMetadataV1,
+}
```

Comment:
@matklad do we need a enum here? I think it might be better to just have a struct and have custom borsh serializer/deserializer

Reply:
We do need an enum, exactly to make this migration the only required. In the future, we'll be able to add new bits of data to Metadata in a compatible, migration less way.

Maybe I am missing something, but I don't see how a custom serializer can help us here. `ExecutionOutcome` is a borsh-serialized struct, we can't look at the bytes of serialized execution outcome and figure out if it has an extra field in the end or not?

Reply:
I was thinking that maybe we could try to deserialize it into the new one first and if it does not work we deserialize it into the old one, but that might not actually work now that I think about it again :(